src/main/java/com/google/sps/servlets/InfoCardServlet.java

```diff
@@ -0,0 +1,155 @@
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // find the corresponding election and check fields
+    // TODO (anooshree): iterate through Datastore and find the election
+    //                   with the matching electionId
+    // if this election is already populated, we don't need to make another
+    // query
+
+    StringBuilder strBuf = new StringBuilder();
+    HttpURLConnection conn = null;
+    BufferedReader reader = null;
+
+    try {
+      String address = request.getParameter("address");
+      if (address == null) {
+        response.setContentType("text/html");
+        response
+            .getWriter()
+            .println("No address in the query URL, please check why this is the case.");
+        return;
+      }
+
+      String electionId = request.getParameter("electionId");
+      if (electionId == null) {
+        response.setContentType("text/html");
+        response
+            .getWriter()
+            .println("No election ID in the query URL, please check why this is the case.");
+        return;
+      }
+
+      URL url =
+          new URL(
+              String.format(
+                  "%s?address=%s&electionId=%s&key=%s",
+                  BASE_URL,
+                  address,
+                  electionId,
+                  getApiKey("112408856470", "election-api-key", "1")));
+      conn = (HttpURLConnection) url.openConnection();
+
+      conn.setRequestMethod("GET");
+      conn.setRequestProperty("Accept", "application/json");
+
+      if (conn.getResponseCode() != HttpURLConnection.HTTP_OK) {
+        throw new RuntimeException(
+            "HTTP GET Request Failed with Error code : " + conn.getResponseCode());
+      }
+
+      // Using IO Stream with Buffer for increased efficiency
+      reader = new BufferedReader(new InputStreamReader(conn.getInputStream(), "utf-8"));
+      String output = null;
+
+      while ((output = reader.readLine()) != null) {
+        strBuf.append(output);
+      }
+    } catch (MalformedURLException e) {
+      response.setContentType("text/html");
+      response.getWriter().println("URL is incorrectly formatted");
+      return;
+    } catch (IOException e) {
+      response.setContentType("text/html");
+      response.getWriter().println("Cannot retrieve information from provided URL");
+      return;
+    } finally {
+      if (reader != null) {
+        try {
+          reader.close();
+        } catch (IOException e) {
+          logger.log(Level.WARNING, e.getMessage());
+        }
+      }
+      if (conn != null) {
+        conn.disconnect();
+      }
+    }
+
+    String results = strBuf.toString();
+    JSONObject obj = new JSONObject(results);
+
+    // TODO(anooshree, caseyprice): process JSON objects from API to store as
+    //                              Datastore entities using decomposed functions
+
+    /*JSONArray pollingLocationData = obj.getJSONArray("pollingLocations");
```

Comment:
Commented out code is generally pretty discouraged:
https://www.informit.com/articles/article.aspx?p=1334908